ee/tables/tablehelpers/exec.go

```diff
@@ -6,14 +6,57 @@ import (
 	"fmt"
 	"log/slog"
 	"os"
+	"os/exec"
+	"os/user"
 	"path/filepath"
+	"strconv"
+	"syscall"
 	"time"
 
 	"github.com/kolide/launcher/ee/allowedcmd"
 	"github.com/kolide/launcher/pkg/traces"
 	"go.opentelemetry.io/otel/attribute"
 )
 
+type execOps func(*exec.Cmd) error
+
+func WithUid(uid string) execOps {
+	return func(cmd *exec.Cmd) error {
+		currentUser, err := user.Current()
+		if err != nil {
+			return fmt.Errorf("getting current user: %w", err)
+		}
+
+		runningUser, err := user.LookupId(uid)
+		if err != nil {
+			return fmt.Errorf("looking up user with uid %s: %w", uid, err)
+		}
+
+		if currentUser.Uid != "0" && currentUser.Uid != runningUser.Uid {
+			return fmt.Errorf("current user %s is not root and can't start process for other user %s", currentUser.Uid, uid)
+		}
+
+		runningUserUid, err := strconv.ParseUint(runningUser.Uid, 10, 32)
+		if err != nil {
+			return fmt.Errorf("converting uid %s to int: %w", runningUser.Uid, err)
+		}
+
+		runningUserGid, err := strconv.ParseUint(runningUser.Gid, 10, 32)
+		if err != nil {
+			return fmt.Errorf("converting gid %s to int: %w", runningUser.Gid, err)
+		}
+
+		cmd.SysProcAttr = &syscall.SysProcAttr{
+			Credential: &syscall.Credential{
```

Comment:
I think you'll need an exec_posix.go and an exec_windows.go to be able to define `WithUid` separately

Reply:
Broke it out into a different go file